yt/yt/server/lib/exec_node/config.cpp

```diff
@@ -878,6 +890,18 @@ void TExecNodeConfig::Register(TRegistrar registrar)
 
     registrar.Parameter("job_proxy", &TThis::JobProxy)
         .DefaultNew();
+
+    registrar.Parameter("job_proxy_log_manager", &TThis::JobProxyLogManager)
+        .Default();
+
+    registrar.Postprocessor([] (TExecNodeConfig* config) {
+        if (config->JobProxy->JobProxyLogging->Mode == EJobProxyLoggingMode::Simple) {
+            return;
+        }
+        if (config->JobProxyLogManager == nullptr) {
+            THROW_ERROR_EXCEPTION("\"per_job_directory\" logging mode requires \"job_proxy_log_manager\" to be set");
```

Comment:
I think that you should also check that `directory` and `sharding_key_length` is specified.